### PR TITLE
Improvement/connection manager refactoring

### PR DESCRIPTION
**Pull request by EmiM**

### CODE REVIEW

File: src/libp2p/connectionsManager.ts [hunk 1]
```diff
@@ -138,39 +147,41 @@ export class ConnectionsManager {
   }
 
   public initializeNode = async (staticPeerId?: PeerId): Promise<ILibp2pStatus> => {
+    console.log('local address:', this.localAddress)
+    console.log('bootstrapMultiaddrs:', this.bootstrapMultiaddrs)
```
Comment (vinkabuki) [inline src/libp2p/connectionsManager.ts:151]: Shouldn't we use log instead console.log?

File: src/libp2p/connectionsManager.ts [hunk 2]
```diff
@@ -138,39 +147,41 @@ export class ConnectionsManager {
...
+
     if (!staticPeerId) {
       this.peerId = await this.getPeerId()
     } else {
       this.peerId = staticPeerId
     }
     this.createAgent()
+    this.localAddress = `${this.listenAddrs}/p2p/${this.peerId.toB58String()}`
```
Comment (vinkabuki) [inline src/libp2p/connectionsManager.ts:159]: linter won't allow using array in string expression, but it's already fixed no 'add-test' branch

Comment (EmiM) [reply src/libp2p/connectionsManager.ts:159]: fixed

File: src/libp2p/connectionsManager.ts [hunk 3]
```diff
@@ -14,25 +14,27 @@ import { ZBAY_DIR_PATH } from '../constants'
...
-interface IOptions {
-  env?: {
-    appDataPath: string
-  },
-  bootstrapMultiaddrs?: Array<string>
+class IOptions {
+  env: {
+    appDataPath?: string
+  } = {}
+  bootstrapMultiaddrs: Array<string> = []
```
Comment (vinkabuki) [inline src/libp2p/connectionsManager.ts]: let's try to be consistent and use either Array<string> or string[] across the whole project, but not both

File: src/libp2p/connectionsManager.ts [hunk 4]
```diff
@@ -58,6 +60,7 @@ export class ConnectionsManager {
   socksProxyAgent: any
   libp2p: null | CustomLibp2p
   localAddress: string | null
+  listenAddrs: string[]
```
Comment (vinkabuki) [inline src/libp2p/connectionsManager.ts]: let's try to be consistent and use either Array or string[] across the whole project, but not both

**Pull request merged**